src/utils.ts

```diff
@@ -0,0 +1,14 @@
+type WithHeaders = { headers: unknown };
+
+// TODO: do not use `any`
+export const merge = (...objs: any[]) => {
+  const headers = objs
+    .filter((obj): obj is WithHeaders =>
+      typeof obj === 'object' && obj ? Boolean(obj.headers) : false,
```

Comment:
I think if `typeof obj === 'object'` statement is true we do not need the `&& obj` because it would be redundant.

Maybe `typeof obj === 'object' && !!obj.headers`?

Reply:
`typeof null` is `object` 😕

Reply:
```suggestion
      typeof obj === 'object' && obj ? !!obj.headers : false,
```

I normally do casting with `Boolean` than `!!`, personal preference. But `!!` is shorter so build will be smaller :trollface:

Reply:
Yeah, I prefer `!!` because of performance concerns 😅. Jokes aside I do not mind the `Boolean` I just used `!!` in the example because I am used to that.